applications/adv_networking_bench/cpp/default_bench_op_tx.h

```diff
@@ -131,6 +131,30 @@ class AdvNetworkingBenchDefaultTxOp : public Operator {
       cudaMemcpy(gds_header_, reinterpret_cast<void*>(&pkt), sizeof(pkt), cudaMemcpyDefault);
     }
 
+    bool src_port_is_range = udp_src_port_str_.get().find('-') != std::string::npos;
+    bool dst_port_is_range = udp_dst_port_str_.get().find('-') != std::string::npos;
+    if (src_port_is_range || dst_port_is_range) {
+      // If UDP port is a range, we must ensure that either GPUDirect is disabled
+      // or Header-Data Split (HDS) is enabled (hds > 0).
+      // We make a static L4 header in the packet on the device so it needs to be a constant value.
```

Comment:
so it's not that it can't be done, just that the current app doesn't want to udpate the header for convenience?

Reply:
right, we would have to update GPU memory on each loop or make a bunch of buffers with headers